blockchain/utxoentry_test.go

```diff
@@ -0,0 +1,268 @@
+		txType:    stake.TxTypeRegular,
+		want:      utxoFlags(0x00),
+	}, {
+		name:      "coinbase, has expiry, vote tx",
+		coinbase:  true,
+		spent:     false,
+		modified:  false,
+		hasExpiry: true,
+		txType:    stake.TxTypeSSGen,
+		want:      utxoFlags(0x29),
+	}, {
+		name:      "spent, modified, has expiry, ticket tx",
+		coinbase:  false,
+		spent:     true,
+		modified:  true,
+		hasExpiry: true,
+		txType:    stake.TxTypeSStx,
+		want:      utxoFlags(0x1E),
+	}}
+
+	for _, test := range tests {
+		got := encodeUtxoFlags(test.coinbase, test.spent, test.modified,
+			test.hasExpiry, test.txType)
+		if got != test.want {
+			t.Errorf("%q: unexpected result -- got %x, want %x", test.name, got,
+				test.want)
+		}
+	}
+}
+
+// TestIsTicketSubmissionOutput validates that the determination of whether
+// various inputs represent a ticket submission output or not is correct.
+func TestIsTicketSubmissionOutput(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		txType   stake.TxType
+		txOutIdx uint32
+		want     bool
+	}{{
+		name:     "ticket tx, output 0",
+		txType:   stake.TxTypeSStx,
+		txOutIdx: 0,
+		want:     true,
+	}, {
+		name:     "ticket tx, output 1",
+		txType:   stake.TxTypeSStx,
+		txOutIdx: 1,
+		want:     false,
+	}, {
+		name:     "vote tx, output 0",
+		txType:   stake.TxTypeSSGen,
+		txOutIdx: 0,
+		want:     false,
+	}}
+
+	for _, test := range tests {
+		got := isTicketSubmissionOutput(test.txType, test.txOutIdx)
+		if got != test.want {
+			t.Errorf("%q: unexpected result -- got %v, want %v", test.name, got,
+				test.want)
+		}
+	}
+}
+
+// TestUtxoEntry validates that the utxo entry methods behave as expected under
+// a variety of conditions.
+func TestUtxoEntry(t *testing.T) {
+	t.Parallel()
+
+	// Define constants for indicating flags.
+	const (
+		noCoinbase   = false
+		withCoinbase = true
+		unspent      = false
+		spent        = true
+		unmodified   = false
+		modified     = true
+		noExpiry     = false
+		withExpiry   = true
+	)
+
+	// Create a ticket submission utxo entry.
+	amount := int64(4294959555)
+	pkScript := hexToBytes("ba76a914a13afb81d54c9f8bb0c5e082d56fd563ab9b359688ac")
+	blockHeight := uint32(55555)
+	blockIndex := uint32(1)
+	scriptVersion := uint16(0)
+	ticketMinOuts := hexToBytes("03808efefade57001aba76a914a13afb81d54c9f8bb0c5" +
+		"e082d56fd563ab9b359688ac0000206a1e9ac39159847e259c9162405b5f6c8135d2c7ea" +
+		"f1a375040001000000005800001abd76a914000000000000000000000000000000000000" +
+		"000088ac")
+	deserializedTicketMinOuts := []*stake.MinimalOutput{{
+		PkScript: pkScript,
+		Value:    amount,
+		Version:  scriptVersion,
+	}, {
+		PkScript: hexToBytes("6a1e9ac39159847e259c9162405b5f6c8135d2c7eaf1a375040" +
+			"0010000000058"),
+		Value:   0,
+		Version: scriptVersion,
+	}, {
+		PkScript: hexToBytes("bd76a914000000000000000000000000000000000000000088a" +
+			"c"),
+		Value:   0,
+		Version: scriptVersion,
+	}}
+	ticketSubmissionEntry := &UtxoEntry{
```

Comment:
It's definitely not something necessary for this PR, but I suspect something to keep in mind for the future is potentially converting this over to the normal table-driven test format so multiple can be added.  That way different combinations can more easily be tested when the cache comes along.

Reply:
I think that is a good call, and I updated it.  I originally did not since some of the checks would be pretty redundant across multiple examples.  But since they are really fast, it doesn't really matter, and this will make it simpler to add more cases moving forward.